reposerver/repository/repository.proto

```diff
@@ -36,6 +36,8 @@ message ManifestRequest {
     repeated string projectSourceRepos = 24;
     // This is used to surface "source not permitted" errors for Helm repositories
     string projectName = 25;
+    // This is used during multi-source rollback to replace versions in "ref" sources
+    bool isRollback = 26;
```

Comment:
I'm not sure I'm a fan of introducing the "rollback" concept to the repo-server. Can we instead just collect the sync configuration in the controller or API server and pass that to the repo-server as if we were just syncing to that commit?

Reply:
Let me check it

Reply:
I have removed the logic from here and I have moved it into the state